src/packages/core/workspace/components/workspace-action/shared/workspace-action-menu/workspace-action-menu.element.ts

```diff
@@ -0,0 +1,148 @@
+import type { CSSResultGroup } from '@umbraco-cms/backoffice/external/lit';
+import { css, html, customElement, property, state, repeat, nothing } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import type { ManifestTypes, ManifestWorkspaceActionMenuItem } from '@umbraco-cms/backoffice/extension-registry';
+import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import type { UmbExtensionElementInitializer } from '@umbraco-cms/backoffice/extension-api';
+import { UmbExtensionsElementInitializer } from '@umbraco-cms/backoffice/extension-api';
+import { UmbLitElement } from '@umbraco-cms/backoffice/lit-element';
+import { UMB_WORKSPACE_CONTEXT } from '@umbraco-cms/backoffice/workspace';
+import type { UUIInterfaceColor, UUIInterfaceLook } from '@umbraco-cms/backoffice/external/uui';
+
+@customElement('umb-workspace-action-menu')
+export class UmbWorkspaceActionMenuElement extends UmbLitElement {
+	#workspaceContext?: typeof UMB_WORKSPACE_CONTEXT.TYPE;
+	#actionsInitializer?: UmbExtensionsElementInitializer<ManifestTypes, 'workspaceActionMenuItem'>;
+
+	/**
+	 * The workspace actions to filter the available actions by.
+	 * @example ['Umb.WorkspaceAction.Document.Save', 'Umb.WorkspaceAction.Document.SaveAndPublishNew']
+	 */
+	@property({ type: Array })
+	workspaceActions: Array<string> = [];
+
+	@property()
+	look: UUIInterfaceLook = 'secondary';
+
+	@property()
+	color: UUIInterfaceColor = 'default';
+
+	@state()
+	private _actions: Array<UmbExtensionElementInitializer<ManifestWorkspaceActionMenuItem, never>> = [];
+
+	@state()
+	_popoverOpen = false;
+
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_WORKSPACE_CONTEXT, (context) => {
+			this.#workspaceContext = context;
+			this.#initialise();
+		});
+	}
+
+	disconnectedCallback(): void {
+		super.disconnectedCallback();
+		this.#actionsInitializer?.destroy();
+	}
+
+	#initialise() {
+		if (!this.#workspaceContext) throw new Error('No workspace context');
+
+		// If there are no workspace action aliases, then there is no need to initialize the actions.
+		if (!this.workspaceActions.length) return;
+
+		const unique = this.#workspaceContext.getUnique();
+		const entityType = this.#workspaceContext.getEntityType();
+
+		this.#actionsInitializer = new UmbExtensionsElementInitializer(
+			this,
+			umbExtensionsRegistry,
+			'workspaceActionMenuItem', // TODO: Stop using string for 'workspaceActionMenuItem', we need to start using Const.
+			(action) => {
+				const containsAlias = action.meta.workspaceActions.some((alias) => this.workspaceActions.includes(alias));
+				const isValidEntityType = !action.meta.entityTypes.length || action.meta.entityTypes.includes(entityType);
+				return containsAlias && isValidEntityType;
+			},
+			(ctrls) => {
+				ctrls.forEach((ctrl) => {
+					ctrl.properties = { unique, entityType };
```

Comment:
I see what you are doing here.

You can 'just' set the properties prop on the this.#actionsInitializer.

Like this:
`this.#actionsInitializer.properties = { unique, entityType };`

then it will be parsed to all extension controllers, same effect, but there is a feature for it already. :-)

Reply:
I figured they would be set before initialization this way, but not sure it makes a difference, so I'm going to change it :)